### PR TITLE
National report pdf page 4

### PR DESCRIPTION
**Pull request by tinashechiraya**

### CODE REVIEW

**Comment by tinashechiraya** [comment]
[national_report-21.pdf](https://github.com/kartoza/sawps/files/12450646/national_report-21.pdf)

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 27 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 27 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 29 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 29 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 29 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 29 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 22 min

**Comment by tinashechiraya** [comment]
[national_report-18.pdf](https://github.com/kartoza/sawps/files/12508355/national_report-18.pdf)

with updated styling on pdf

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 21 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 22 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 22 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 22 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 25 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 27 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 25 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 27 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 25 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 29 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 27 min

File: django_project/frontend/tests/test_national_statistics.py [hunk 1]
```diff
@@ -264,3 +264,122 @@ def test_get_activity_count(self):
...
+        )
+        self.auth_headers = {
+            "HTTP_AUTHORIZATION": "Basic "
+            + base64.b64encode(b"testuser:testpassword").decode("ascii"),
+        }
+
+        session = self.client.session
+        session.save()
+        url = self.url
+        response = self.client.get(url, **self.auth_headers)
```
Comment (dimasciput) [inline django_project/frontend/tests/test_national_statistics.py:324]: Please also test  the response data

File: django_project/frontend/tests/test_national_statistics.py [hunk 2]
```diff
@@ -264,3 +264,122 @@ def test_get_activity_count(self):
...
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```
Comment (dimasciput) [inline django_project/frontend/tests/test_national_statistics.py:385]: Add response data testing here

File: django_project/frontend/api_views/national_statistic.py [hunk 3]
```diff
@@ -223,3 +222,169 @@ def get(self, *args, **kwargs) -> Response:
...
+        Get activity count of the total
+        population as percentage
+        """
+        # Step 1: Get all owned species and filter them
+        owned_species = OwnedSpecies.objects.select_related(
+            'taxon', 'property__province'
+        )
+
+        # Step 2: Get all provinces
+        # provinces = Province.objects.all()
```
Comment (dimasciput) [inline django_project/frontend/api_views/national_statistic.py:246]: If not used please remove it

File: django_project/frontend/api_views/national_statistic.py [hunk 4]
```diff
@@ -223,3 +222,169 @@ def get(self, *args, **kwargs) -> Response:
...
+        """
+        # Step 1: Get all owned species and filter them
+        owned_species = OwnedSpecies.objects.select_related(
+            'taxon',
+            'property__province',
+            'property__property_type'
+        )
+
+        # Step 2: Get all provinces
+        # provinces = Province.objects.all()
```
Comment (dimasciput) [inline django_project/frontend/api_views/national_statistic.py:330]: Please remove

**Review by dimasciput — CHANGES_REQUESTED** [review]

**Review by dimasciput — APPROVED** [review]

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 22 min

**Pull request merged**